my-app/src/Logic.js

```diff
@@ -42,49 +43,84 @@ return checked;
 
 }
 
+function score(state, erase){
+  state.whiteIsNext
+  ? state.blackScore += erase.length
+  : state.whiteScore += erase.length
+
+  wipe(state, erase);
+}
+
+function wipe(state, erase){
+  for(let i=0;i < erase.length;i++){
+    erase[i].value = 'blank'
+  }
+}
+
 function getNeighborsOfValue(state, cell, value){
   let neighbors = []
+  let x = cell.x
+  let y = cell.y
+  let board = state.gameBoard.board
 
-  if(state.gameBoard.board[cell.x][cell.y+1].value === value){
-    neighbors.push(state.gameBoard.board[cell.x][cell.y+1])
+  if(board[x][y+1]){
+    if(board[x][y+1].value === value){
+      neighbors.push(board[x][y+1])
+    }
   }
-  if(state.gameBoard.board[cell.x+1][cell.y].value === value){
-    neighbors.push(state.gameBoard.board[cell.x+1][cell.y])
+  if(board[x+1]){
+    if(board[x+1][y].value === value){
+      neighbors.push(board[x+1][y])
+    }
   }
-  if(state.gameBoard.board[cell.x][cell.y-1].value === value){
-    neighbors.push(state.gameBoard.board[cell.x][cell.y-1])
+  if(board[x][y-1]){
+    if(board[x][y-1].value === value){
+      neighbors.push(board[x][y-1])
+    }
   }
-  if(state.gameBoard.board[cell.x-1][cell.y].value === value){
-    neighbors.push(state.gameBoard.board[cell.x-1][cell.y])
+  if(board[x-1]){
+    if(board[x-1][y].value === value){
+      neighbors.push(board[x-1][y])
+    }
   }
-
   return neighbors
 
 }
 
 function difference(arr1, arr2){
-  arr1 = arr1.concat(arr2);
-  return arr1.filter(function(val){
-   return arr1.indexOf(val) === arr1.lastIndexOf(val);
-  });
-
+  let newArr = []
```

Comment:
You should use intention revealing names - `arr1` and `arr2` do not reveal the intent of the variable.  As well, this could be replaced by a more concise functional approach (`filter`).